### PR TITLE
Create a task

### PR DESCRIPTION
**Pull request by colmdoyle**
###  Summary

Put in a basic scaffold for how to add a new shortcut

### Requirements (place an `x` in each `[ ]`)

* [X] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/tasks-app/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [X] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: nodejs/listeners/shortcuts/global-new-task.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    app.shortcut('global_new_task', async ({ shortcut, ack, client }) => {
+        try {
+            await ack();
+            await client.views.open({
+                trigger_id: shortcut.trigger_id,
+                view: modals.newTaskModal()
+            });
+            console.log('global_new_task called')
+        } catch (error) {
+            console.error(error);
```
[the remark this answers is not in the document]
Comment (seratch) [reply nodejs/listeners/shortcuts/global-new-task.js:14]: That's true. `views.publish` may work in some scenario (it'd be quite rare situation though) but it won't be so useful.

File: nodejs/migrations/20210614114545-create-task.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        type: Sequelize.STRING
+      },
+      isComplete: {
+        type: Sequelize.BOOLEAN
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      updatedAt: {
```
[the remark this answers is not in the document]
Comment (seratch) [reply nodejs/migrations/20210614114545-create-task.js:21]: Okay, I got it!

File: nodejs/migrations/20210614115427-create-user.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      slackUserID: {
+        type: Sequelize.STRING,
+        allowNull: false,
+      },
+      slackOrganizationID: {
+        type: Sequelize.STRING
+      },
+      slackWorkspaceID: {
+        type: Sequelize.STRING,
+        allowNull: false,
```
Comment (seratch) [inline nodejs/migrations/20210614115427-create-user.js:20]: I should mention this a bit more clearly in the previous comment but what do you think about the user management in this table for org-wide installation? 

If the app works as an org-wide app, the app does not need to know workspace IDs. Therefore, this column can be nullable if we support org-wide. Also, storing something equivalent to `is_enterprise_install: boolean` in this table may allow the app to more easily handle both single workspace installations and org-wide installations.

Another idea would be to have a bit different table like `OrgUsers` for org-wide installation patterns. But, probably, this way will make the app implementation more complex.

**Review by seratch — APPROVED** [review]
Left a few more comments but this is just the beginning of the development. If you update some parts later on, go ahead with the current implementation for now.

File: nodejs/migrations/20210614115427-create-user.js (hunk 3 above)
Comment (colmdoyle) [reply nodejs/migrations/20210614115427-create-user.js:20]: We'll definitely want to make the app org-aware, but will handle it in a later PR, as it'll give us a chance to walk through a migration one style from to the other.

Comment (seratch) [reply nodejs/migrations/20210614115427-create-user.js:20]: That makes sense 👍

**Comment by mwbrooks** [comment]
I'm slow on leaving feedback for this PR. Sorry about that!

The `views`, `block-kit`, `templates` discussion is interesting and something I've faced myself a few times. Right now, I organize my personal apps with:

* `templates/`
    * Exports the raw Block Kit JSON
    * Exports a render function that injects values into parts of template
    * Exports the commonly referenced Ids, such as `viewCallbackId` or `taskTitleBlockId`
* `views/`
    * Imports and renderes a template
    * Uses the exported Ids to easily reference parts of the view, like getting the task title content

There are pros-and-cons to this approach, but it's helped keep the code organized, readable, and flexible so far.

**Pull request merged**